File: containerWithMostWater_BruteForce.java
// Time Complexity : O(n^2) where n is the length of the array nums
// Space Complexity : O(1) 
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No

// Your code here along with comments explaining your approach
// for every num in array, find area with every possible number.
// output max

class containerWithMostWater_BruteForce {
    public int maxArea(int[] height) {
        int n = height.length;
        int ans = 0;
        for (int i = 0; i < n; i++) {
            for (int j = n-1; j > i; j--) {
                ans = Math.max(ans, (j-i) * Math.min(height[i], height[j]));
            }
        }
        return ans;
    }
}

File: Container_with_Most_Water_BruteForce.py
# Created by Aashish Adhikari at 9:03 AM 1/9/2021

'''
Time Complexity:
O(n^2)
Space Complexity:
O(1)
'''

class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """

        maxx = 0
        for idx in range(len(height)):

            for idxx in range(idx+1, len(height)):
                area = min(height[idx], height[idxx]) * (idxx - idx)
                if area > maxx:
                    maxx = area

        return maxx